### PR TITLE
Vanity Name Servers

### PR DESCRIPTION
**Pull request by jodosha**
Added support for:

  * `enableVanityNameServers`
  * `disableVanityNameServers`

@weppos @aeden Please review.

### CODE REVIEW

**Comment by weppos** [comment]
👍

**Comment by weppos** [comment]
Actually, one thing... instead of having a separate service, I think they should be under the NameServers service.

File: dnsimple/vanity_name_server.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func vanityNameServerPath(accountID string, domainID string) string {
+	return fmt.Sprintf("/%v/vanity/%v", accountID, domainID)
+}
+
+// Enable Vanity Name Servers for the given domain
+//
+// See https://developer.dnsimple.com/v2/vanity/#enable
+func (s *VanityNameServersService) Enable(accountID string, domainID string) (*VanityNameServerResponse, error) {
```
Comment (weppos) [inline dnsimple/vanity_name_server.go]: It should be EnableVanityNameServers. Same for Disable. We have been using qualified names to avoid conflicts.

**Comment by jodosha** [comment]
@weppos I'm reflecting the "groups" that we have for API v2 documentation. Shall we move VNS under NS both in docs and code?

File: dnsimple/vanity_name_server.go (hunk 1 above)
Comment (jodosha) [inline dnsimple/vanity_name_server.go]: Okay 👍

**Comment by weppos** [comment]
That's a good question. Changing the delegation of a name server should probably stay under Delegation.
https://developer.dnsimple.com/v2/registrar/delegation/

I'm actually torn about the name servers. I'd like to keep the number of services limited, especially for cases where we have a very limited set of actions. It's also true that we don't have a lot of other places where to keep this feature for now.

Let's keep it as it is, until we have a global overview of the final API actions.

File: dnsimple/vanity_name_server.go (hunk 1 above)
Comment (jodosha) [inline dnsimple/vanity_name_server.go]: https://github.com/dnsimple/dnsimple-go/commit/469860ab74b29df80665a68a081f1745f48b6888

**Pull request merged**